sdc/tests/tests_perf/test_perf_df.py

```diff
@@ -0,0 +1,100 @@
+#     this list of conditions and the following disclaimer in the documentation
+#     and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+
+import pandas as pd
+import numpy as np
+
+import time
+import random
+
+import pandas
+import sdc
+
+from sdc.tests.tests_perf.test_perf_base import TestBase
+from sdc.tests.tests_perf.test_perf_utils import calc_compilation, get_times, perf_data_gen_fixed_len
+from sdc.tests.test_utils import test_global_input_data_float64
+
+
+def usecase_df_min(input_data):
+    start_time = time.time()
+    res = input_data.min()
+    finish_time = time.time()
+
+    return finish_time - start_time, res
+
+
+# python -m sdc.runtests sdc.tests.tests_perf.test_perf_df.TestDataFrameMethods
+class TestDataFrameMethods(TestBase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.total_data_length = {
+            'min': [10 ** 7],
+        }
+
+    def _test_jitted(self, pyfunc, record, *args, **kwargs):
+        # compilation time
+        record["compile_results"] = calc_compilation(pyfunc, *args, **kwargs)
+
+        sdc_func = sdc.jit(pyfunc)
+
+        # Warming up
+        sdc_func(*args, **kwargs)
+
+        # execution and boxing time
+        record["test_results"], record["boxing_results"] = \
+            get_times(sdc_func, *args, **kwargs)
+
+    def _test_python(self, pyfunc, record, *args, **kwargs):
+        record["test_results"], _ = \
+            get_times(pyfunc, *args, **kwargs)
+
+    def _test_case(self, pyfunc, data_name, test_name=None, input_data=test_global_input_data_float64):
+        test_name = test_name or data_name
+
+        full_input_data_length = sum(len(i) for i in input_data)
+        for data_length in self.total_data_length[data_name]:
+            base = {
+                "test_name": test_name,
+                "data_size": data_length,
+            }
+            data = perf_data_gen_fixed_len(input_data, full_input_data_length,
+                                           data_length)
+            test_data = pandas.DataFrame({column: data for column in ('A', 'B', 'C')})
```

Comment:
You could use `string.ascii_uppercase` from default module `string` to get all the letters and generate more big `DataFrame`. What do you think about that?